src/test/java/fi/metatavu/soteapi/wnspusher/WnsPusherNotificationControllerTest.java

```diff
@@ -0,0 +1,159 @@
+                  .getValue()
+                  .getLastHeader("Authorization")
+                  .getValue(),
+                 "APIKEY TEST_API_KEY");
+  }
+
+  @Test
+  public void testSendNotificationApp()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+    assertEquals(notification.get("app").asText(), "TEST_APP");
+  }
+
+  @Test
+  public void testSendNotificationUrl()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    HttpPost httpPost = httpClientArguments.getValue();
+    assertEquals(httpPost.getURI().toString(), "TEST_ENDPOINT");
+  }
+
+  @Test
+  public void testSendNotificationContentType()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+    assertEquals(notification.get("contentType").asText(), "application/xml");
+  }
+
+  @Test
+  public void testSendNotificationWnsType()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+    assertEquals(notification.get("wnsType").asText(), "wns/toast");
+  }
+
+  @Test
+  public void testSendNotificationContent()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+
+    String expectedXml = "<toast launch=\"null\"><visual>" +
+                         "<binding template=\"ToastText02\">" + 
+                         "<text id=\"1\">TEST_TITLE</text>" + 
+                         "<text id=\"2\">TEST_MESSAGE</text></binding>" + 
+                         "</visual></toast>";
+    assertEquals(notification.get("content").asText(), expectedXml);
+  }
+
+  @Test
+  public void testSendNotificationNonAsciiChars()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)